dlt/destinations/dataset/dataset.py

```diff
@@ -193,26 +210,16 @@ def table(self, table_name: str) -> ReadableIbisRelation:
                 f" {', '.join(self.schema.tables.keys())}"
             )
 
-        # we can create an ibis powered relation if ibis is available
-        relation: BaseReadableDBAPIRelation
-        if self._dataset_type == "ibis":
+        if table_type == "ibis":
             from dlt.helpers.ibis import create_unbound_ibis_table
-            from dlt.destinations.dataset.ibis_relation import ReadableIbisRelation
 
-            unbound_table = create_unbound_ibis_table(self.schema, self.dataset_name, table_name)
-            relation = ReadableIbisRelation(
-                readable_dataset=self,
-                ibis_object=unbound_table,
-            )
+            return create_unbound_ibis_table(self.schema, self.dataset_name, table_name)
 
-        else:
-            # fallback to the standard dbapi relation
-            relation = ReadableDBAPIRelation(
-                readable_dataset=self,
-                table_name=table_name,
-            )
-
-        return relation  # type: ignore[return-value]
+        # fallback to the standard dbapi relation
+        return ReadableDBAPIRelation(
+            readable_dataset=self,
+            table_name=table_name,
+        )
 
     def row_counts(
```

Comment:
what should we do with this? I was thinking if we could inject additional methods into dataset? like query packages ie. dataset.stats.row_count. or maybe we move this to queries? this is not public interface now, right?

Reply:
Row counts is a public interface and it is documented already since we released the dataset. But I agree that it would be nice to have a place for special queries such as @zilto will need for the incremental datasets where you can filter by load_id etc. Maybe let's keep it here for now and move it once we know how to access these special queries?